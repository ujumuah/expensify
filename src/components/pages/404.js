import React from "react"
import {Link} from "react-router-dom"
 
const a =()=> <div><h1>Opps!404,Page not found-<Link to="/">Go home</Link></h1></div>;

export default a;