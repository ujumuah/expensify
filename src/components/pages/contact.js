import React from "react"
 const a =()=>(
     <div>
        <h1>Contact  jumauh uthman@gmial.com-A reactive developer</h1>
        <h2>Contact number: +101 23254478</h2>
     </div>
 );
 export  default a;