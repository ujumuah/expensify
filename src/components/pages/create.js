import  React from 'react'
import {connect} from "react-redux"
import {addExpense}  from '../../redux-sample/actionGen/expenses'
import  Expenseform from './expenseForm'
     export class AddExpensePage extends React.Component{
      constructor(props){
          super(props);
          this.FormSubmit = this.FormSubmit.bind(this)
      }
      FormSubmit(expenses){
          // There is an high tendency that the error 
          // is coming from the disptch function
           this.props.FormSubmit(expenses)
          this.props.history.push('/')
      }
      render(){
          return (
            <div>
            <h1>ADD EXPENSES</h1>
              <Expenseform FormSubmit={this.FormSubmit}/>
       </div>
          )
      }
    }
 
const matchDispatchToprop =(dispatch)=>({
    FormSubmit:(expenses)=>dispatch(addExpense(expenses))
})
 
export default connect(undefined,matchDispatchToprop)(AddExpensePage);