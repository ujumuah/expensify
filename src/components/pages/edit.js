import  React from 'react'
import {connect} from "react-redux"
import { editExpense}  from '../../redux-sample/actionGen/expenses'
import { removeExpense}  from '../../redux-sample/actionGen/expenses'
import  Expenseform from './expenseForm'
 export class A extends React.Component{
     constructor(props){
         super(props);
            this.editExpense = this.editExpense.bind(this)
            this.removeExpense = this.removeExpense.bind(this)

        }
        editExpense(expense){
            console.log(this.props)
             this.props.editExpense(this.props.expenses.id,expense)
            this.props.history.push('/')
        }
        removeExpense(){
            this.props.removeExpense(this.props.expenses.id)
            this.props.history.push('/')
        }
     render(){
         return(
            <div>
            <h1>Edit EXPENSES</h1>
        <Expenseform    
        FormSubmit={ 
         this.editExpense}
      expense={this.props.expenses}
      />
      <div><button onClick={(e)=>{
          e.preventDefault()
          this.removeExpense()
       }}>remove</button>
                      </div> 
          </div>
         )
     }
 }
const matchDispatchToProps =(dispatch)=>(
   {
       editExpense:(id,expense)=>dispatch(editExpense(id,expense)),
       removeExpense:(id)=>{dispatch(removeExpense({id}))}
   } 
    
),
matchToProp =(state,props)=>{
const o = {
  expenses:state.expenses.find((expense)=> expense.id === props.match.params.id)
    }
 return  o
 }
export default connect(matchToProp,matchDispatchToProps)(A);