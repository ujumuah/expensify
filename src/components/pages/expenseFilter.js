import React from 'react'
import {connect} from 'react-redux'
import {textFilter,sortByAmount,sortByDate,startDate as setStartDate,endDate as setEndDate, startDate} from '../../redux-sample/actionGen/filter'
import {DateRangePicker} from 'react-dates'

 export class Text extends React.Component{
   constructor(props){
      super(props);
      this.state={
         calenderFocused:null
      }
      this.onDatesChange= this.onDatesChange.bind(this)
      this.onFocusChange =this.onFocusChange.bind(this)
      this.textChange = this.textChange.bind(this)
      this.selectChange = this.selectChange.bind(this)
   }
   onDatesChange({startDate,endDate}){
       this.props.setStartDate(startDate)
      this.props.setEndDate(endDate)
   }
   onFocusChange(calenderFocused){
      this.setState({calenderFocused})
   }
   textChange(e){
      this.props.textFilter(e.target.value)
   }
   selectChange(e){
       const target = e.target.value
      if(target==='amount'){
          this.props.sortByAmount(target);
      }else if(target==='date')
            this.props.sortByDate(target);
  }
   render(){
      return(
         <div>
         <input type="text"
       value={this.props.filter.text} 
       onChange={this.textChange}/>
       <select value={this.props.filter.sortBy} onChange={this.selectChange}>
         <option value="date">Date</option>
         <option value="amount">Amount</option>
       </select>
          <DateRangePicker
          startDate={this.props.filter.startDate}
          endDate={this.props.filter.endDate} 
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calenderFocused}
          onFocusChange={this.onFocusChange}
         isOutsideRange={()=>false}
         numberOfMonths={1}
         showClearDates={true}
          />
    </div>
     
)
}
}
const mapStateToProp=(state)=>{
   return{ filter:state.filter}
  },
  mapDispatchToprops =(dispatch)=>({
     textFilter:(text)=>dispatch(textFilter(text)),
      setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
      setEndDate:(endDate)=>dispatch(setEndDate(endDate)),
      sortByAmount:(value)=>dispatch(sortByAmount(value)),
      sortByDate:(value)=>dispatch(sortByDate(value))

   })
 

  export default  connect(mapStateToProp,mapDispatchToprops)(Text)