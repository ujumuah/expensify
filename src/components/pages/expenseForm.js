import React ,{Component}from "react"
import  moment from 'moment'
import {SingleDatePicker} from 'react-dates'
// import 'react-dates/lib/css/_datepicker.css'

   class a extends Component{
      constructor(props){
           super(props);
           this.state={
                description: !!props.expense?props.expense.description:'',
               notes:!!props.expense?props.expense.note:'',
               amount:!!props.expense?props.expense.amount:'',
               error:'',
               calenderFocused:false,
                createdAt:!!props.expense?moment(props.expense.createAt):moment()
           }, 
          this.onDescription = this.onDescription.bind(this)
          this.onNote = this.onNote.bind(this)
          this.onAmountChange = this.onAmountChange.bind(this)
           this.onDateChange = this.onDateChange.bind(this)
           this.formSubmit = this.formSubmit.bind(this)
           this.onCalenderFocus = this.onCalenderFocus.bind(this)
       }
       onCalenderFocus({focused}){
         this.setState(()=>({calenderFocused:focused}))
       }
      onDescription(e){
          const description=e.target.value;
          this.setState(()=>({description}))
      }
       onNote(e){
          const notes=e.target.value;
          this.setState(()=>({notes}))
      }
    onAmountChange(e){
        const amount = e.target.value
          if (!amount || /^\d{1,}(\.\d{0,2})?$/.test(amount)){
            this.setState(()=>({amount}))
        }
      }
      onDateChange(createdAt){
       this.setState(()=>({createdAt}))
    }
       formSubmit(e){
           e.preventDefault();
             if(!!!this.state.description&&!!!this.state.amount){
                this.setState(()=>({error:'please insert  description and amount'}))
             }else{
                this.setState(()=>({error:''}))
            this.props.FormSubmit({
                description:this.state.description,
                amount:parseFloat(this.state.amount),
                note:this.state.notes,
                createAt:this.state.createdAt.valueOf(),

            })
            }
       }
        
     
      render(){

          return(
              
              <div>
                 {!!this.state.error&& <h4 className='error'>{this.state.error}</h4>}
              <form onSubmit={(e)=>{this.formSubmit(e)}}>
                <label>
                Description:
                    <input type="text" name='description' placeholder='Description' 
                    value={this.state.description }
                     onChange={(e)=>{this.onDescription(e)}} required='required'/>
                </label>
                <label>
                Amount:
                    <input type="text"
                     name='Amount'
                     onChange={(e)=>{this.onAmountChange(e)}}
                     placeholder='Amount'
                     autoFocus
                     value={this.state.amount}
                     required='required'/>
                </label>
                <SingleDatePicker
                date={this.state.createdAt}
                onDateChange={this.onDateChange}
                focused={this.state.calenderFocused}
                onFocusChange={this.onCalenderFocus}
                numberOfMonths={1}
                isOutsideRange={()=>false}
                 />             
                <textarea  value={this.state.notes}
                required='required'
                 onChange={(e)=>{this.onNote(e)}} placeholder="Explain a litte about the expenses"></textarea>
              <input type='submit' value='Add expense'/>
              </form>
             </div>
          )

      }
  }
  
       
 export  default a;