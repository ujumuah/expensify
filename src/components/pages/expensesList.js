import React from 'react'
import {connect} from 'react-redux'
import  Item from './expensesListItems'
import selectVisibleItem  from '../../redux-sample/selector/visibleStore'
import ExpenseFilter from './expenseFilter' 
  
  export function expenseList(props) {
     return (
        <div>
            <h2>Expenses List</h2>
           <ExpenseFilter/>
             <ul>
                 {
                     props.expenses.length===0?
                        <li>No Expenses</li>
                    :
                props.expenses.map((x) =><Item key={x.id} {...x} />)
                 }
 
            </ul>
        </div>
    )
}
 const mapStateToProp = (state)=>{
        return{
         expenses:selectVisibleItem(state.expenses,state.filter)
     }
 },
  connectExpenseList=connect(mapStateToProp)(expenseList);

 export default connectExpenseList;