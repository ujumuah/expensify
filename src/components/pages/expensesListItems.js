import React from 'react'
import moment from 'moment'
import {connect}from 'react-redux'
import{Link}  from 'react-router-dom'
import {removeExpense} from '../../redux-sample/actionGen/expenses'
const items  =({id,description,amount,createAt})=>(
<div>
     <Link to={`/edit/${id}`}>
    <h4>Expense:{description}</h4>
    </Link>
<div>Amount:{amount}</div>
<div>Created on :{moment(createAt).format('DD-MMM-YYYY')}</div>
</div>

)
export default items;
  