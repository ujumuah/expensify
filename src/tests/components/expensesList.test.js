import React from 'react'
import {shallow} from 'enzyme'
import {expenseList as Expenses} from '../../components/pages/expensesList'
import  expensesSample from  '../expenses/samples'
test('should render expenseList with expensesSamples value',()=>{
 const wrapper  = shallow(<Expenses expenses={expensesSample}/>)
 expect(wrapper).toMatchSnapshot()
})
test('should render expenseList with  No expenses message',()=>{
    const wrapper = shallow(<Expenses expenses={[]}/>)
    expect(wrapper).toMatchSnapshot()
})