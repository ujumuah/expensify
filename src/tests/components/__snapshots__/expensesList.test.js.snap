// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList with  No expenses message 1`] = `
<div>
  <h2>
    Expenses List
  </h2>
  <Connect(Text) />
  <ul>
    <li>
      No Expenses
    </li>
  </ul>
</div>
`;

exports[`should render expenseList with expensesSamples value 1`] = `
<div>
  <h2>
    Expenses List
  </h2>
  <Connect(Text) />
  <ul>
    <items
      amount={1225}
      createAt={0}
      description="rent"
      id="1"
      key="1"
      note="need to pay december rent"
    />
    <items
      amount={1225}
      createAt={0}
      description="mortgage"
      id="2"
      key="2"
      note="need to pay december morgage"
    />
    <items
      amount={2356}
      createAt={0}
      description="Food"
      id="3"
      key="3"
      note="Foood for life"
    />
    <items
      amount={25468}
      createAt={0}
      description="Tour"
      id="4"
      key="4"
      note="need to travel before end of the year"
    />
  </ul>
</div>
`;
